torchvision/csrc/cpu/image/readjpeg_cpu.cpp

```diff
@@ -100,15 +101,35 @@ torch::Tensor decodeJPEG(const torch::Tensor& data) {
 
   // read info from header.
   jpeg_read_header(&cinfo, TRUE);
+
+  int current_channels = cinfo.num_components;
+
+  if (components > 0 && components != current_channels) {
+    switch (components) {
+      case 1: // Gray
+        cinfo.out_color_space = JCS_GRAYSCALE;
+        break;
+      case 3: // RGB
+        cinfo.out_color_space = JCS_RGB;
+        break;
+      default:
```

Comment:
libjpeg does not support converting from CMYK to grayscale etc. There is a way to do this but it [involves converting it](https://github.com/tensorflow/tensorflow/blob/86871065265b04e0db8ca360c046421efb2bdeb4/tensorflow/core/lib/jpeg/jpeg_mem.cc#L284-L313) manually to RGB. Perhaps we can do this on the future.

Reply:
Sounds good to me. Maybe we can add this as a comment in the code?

Reply:
done